src/fake-timers-src.js

```diff
@@ -767,7 +786,11 @@ function withGlobal(_global) {
         clock.nextTick = function nextTick(func) {
             return enqueueJob(clock, {
                 func: func,
-                args: Array.prototype.slice.call(arguments, 1)
+                args: Array.prototype.slice.call(arguments, 1),
+                stack: new Error().stack
```

Comment:
`.stack` is potentially somewhat expensive, could we access it lazily in the error case?

Reply:
@SimenB I'm not sure how we can get the right stack trace unless we generate it when the job is created. Just to see the difference, I added these properties locally (and the same equivalent in `nextTick`):
```js
        timer.fullStack = new Error().stack;
        timer.getStackLazy = function() { return new Error().stack };
```

And then logged them out within `getInfiniteLoopError`:
```js
        console.log(job.fullStack);
        console.log('---');
        console.log(job.getStackLazy());
```

The stack trace generated lazily provides the running stack trace (i.e. `runMicrotasks` or `runAll` calls) rather than the stack trace of the job itself.

❓ Is there a different way we can do it?

<details>
    <summary>Stack trace comparison - first stack is generated when adding the job, second is generated lazily when an infinite loop is detected</summary>



#### Running microtasks
```
Error
      at Object.nextTick (src/fake-timers-src.js:803)
      at Object.queueMicrotask (src/fake-timers-src.js:809)
      at recursiveQueueMicroTask (test/fake-timers-test.js:4639)
      at runJobs (src/fake-timers-src.js:277)
      at Object.runMicrotasks (src/fake-timers-src.js:880)
      at Context.<anonymous> (test/fake-timers-test.js:4648)
---
Error
      at Object.getStackLazy (src/fake-timers-src.js:804)
      at getInfiniteLoopError (src/fake-timers-src.js:156)
      at runJobs (src/fake-timers-src.js:279)
      at Object.runMicrotasks (src/fake-timers-src.js:880)
      at Context.<anonymous> (test/fake-timers-test.js:4648)
```

#### Running all timers
```
Error
      at addTimer (src/fake-timers-src.js:295)
      at Object.setTimeout (src/fake-timers-src.js:769)
      at target.<computed> (src/fake-timers-src.js:595)
      at recursiveCreateTimer (test/fake-timers-test.js:4668)
      at recursiveCreateTimerTimeout (test/fake-timers-test.js:4669)
      at callTimer (src/fake-timers-src.js:450)
      at Object.next (src/fake-timers-src.js:1050)
      at src/fake-timers-src.js:1139
---
Error
      at Object.timer.getStackLazy (src/fake-timers-src.js:296)
      at getInfiniteLoopError (src/fake-timers-src.js:156)
      at src/fake-timers-src.js:1149
```

</details>

Reply:
I meant to still do `new Error()` here, but only access `stack` if we actually wanna throw the error